warrior/Actions/SeleniumActions/browser_actions.py

```diff
@@ -208,23 +212,25 @@ def browser_launch(self, system_name, browser_name="all", type="firefox",
             browser_details = arguments
 
         for browser in browser_list:
-            arguments = Utils.data_Utils.get_default_ecf_and_et(arguments,
-                                                                self.datafile,
-                                                                browser)
+            arguments = Utils.data_Utils.get_default_ecf_and_et(arguments, self.datafile, browser)
+
             if browser_details == {}:
-                browser_details = selenium_Utils.\
-                    get_browser_details(browser, datafile=self.datafile, **arguments)
+                browser_details = selenium_Utils.get_browser_details(browser,
+                                                                     datafile=self.datafile,
+                                                                     **arguments)
             if browser_details is not None:
-                if type == "firefox":
-                    ff_profile = self.browser_object.\
-                        set_firefoxprofile(proxy_ip, proxy_port)
-                if binary != "" and gecko_path != "":
+                if browser_details["gecko_path"] not in [None, False] and \
+                                browser_details["type"] == "firefox":
+                    ff_profile = self.browser_object.set_firefoxprofile(proxy_ip, proxy_port)
                     browser_inst = self.browser_object.open_browser(
                         browser_details["type"], webdriver_remote_url,
                         binary=binary, gecko_path=gecko_path,
                         profile_dir=ff_profile)
                 else:
-                    pNote("Please provide valid path for binary/geckodriver")
```

Comment:
This is supposed to be a failure case, shouldn't even attempt to launch browser here if geckodriver is not found